app/controllers/application_controller.rb

```diff
@@ -20,28 +17,44 @@ def jsonapi_render(collection, filters)
   end
 
   def set_user
-    @current_user = if auth_token.present?
-        User.from_jwt(Authenticator.from_token(auth_token))
-      else
-        nil
-      end
+    return nil unless auth_token.present?
+
+    @current_user = User.where(
+                      id: panoptes.client.authenticated_user_id,
+                      login: panoptes.client.authenticated_user_login
+                    ).first_or_create.tap do |user|
+                      user.display_name = panoptes.client.authenticated_user_display_name
+
+                      # Explicitly set user admin accessor if encoded in JWT
+                      user.admin = panoptes.client.authenticated_admin?
```

Comment:
This won't unset the admin attribute as the `first_or_create.tap` will not save on the `first` execution. I dare say you might need to combine this change and the roles update together possibly using dirty attribute testing after setting roles locally? 
https://apidock.com/rails/ActiveModel/Dirty/changed%3F

Perhaps something like the below to ensure there is only 1 update / save in the controller block (to be considered with the `on_create` block as well)
```
current_user.admin = panoptes.authenticated_admin?
....
current_user.roles = panoptes.roles(current_user.id)
current_user.roles_refreshed_at = Time.now
if current_user.changed?
  current_user.save
end
```

Reply:
I wasn't feeling the idea of using the dirty attr on a first_or_create, so I switched to first_or_initialize and made sure that whether it's a new user (`roles: nil`) or if anything changed that it would get saved. A little verbose, perhaps, with all the tiny methods, lmk what you think.

Reply:
up to you - FWIW the rails framework does offer dirty tracking via active model / active record https://api.rubyonrails.org/classes/ActiveModel/Dirty.html